contrib/go-chi/chi.v5/chi.go

```diff
@@ -36,22 +35,23 @@ func Middleware(opts ...Option) func(next http.Handler) http.Handler {
 				next.ServeHTTP(w, r)
 				return
 			}
-			opts := []ddtrace.StartSpanOption{
-				tracer.SpanType(ext.SpanTypeWeb),
-				tracer.ServiceName(cfg.serviceName),
-				tracer.Tag(ext.HTTPMethod, r.Method),
-				tracer.Tag(ext.HTTPURL, r.URL.Path),
-				tracer.Measured(),
-			}
+			opts := append(cfg.spanOpts, tracer.Measured())
 			if !math.IsNaN(cfg.analyticsRate) {
 				opts = append(opts, tracer.Tag(ext.EventSampleRate, cfg.analyticsRate))
 			}
-			if spanctx, err := tracer.Extract(tracer.HTTPHeadersCarrier(r.Header)); err == nil {
-				opts = append(opts, tracer.ChildOf(spanctx))
-			}
-			opts = append(opts, cfg.spanOpts...)
-			span, ctx := tracer.StartSpanFromContext(r.Context(), "http.request", opts...)
-			defer span.Finish()
+			span, ctx := httptrace.StartRequestSpan(r, cfg.serviceName, "", false, opts...)
+
+			r = r.WithContext(ctx)
+			ww := middleware.NewWrapResponseWriter(w, r.ProtoMajor)
+
+			defer func() {
```

Comment:
```suggestion
			span, ctx := httptrace.StartRequestSpan(r, cfg.serviceName, "", false, opts...)
			r = r.WithContext(ctx)
			ww := middleware.NewWrapResponseWriter(w, r.ProtoMajor)
			defer func() {
```
I think that removing the empty lines really does make sense to make it obvious that this is a chunk which does one focused thing: start and defer the end of a span.

Reply:
Now fixed